nibabel/arrayproxy.py

```diff
@@ -156,8 +139,9 @@ def __init__(self, file_like, spec, mmap=True, keep_file_open=None):
         # Permit any specifier that can be interpreted as a numpy dtype
         self._dtype = np.dtype(self._dtype)
         self._mmap = mmap
-        self._keep_file_open = self._should_keep_file_open(file_like,
-                                                           keep_file_open)
+        # If using indexed_gzip, we use a single ImageOpener. Otherwise, we
+        # create a new ImageOpener on each file access
+        self._persist_opener = openers.HAVE_INDEXED_GZIP
```

Comment:
Main thought here is that we only want to use a single file if we have `indexed_gzip` and the file is a `.gz` file. Quick read-through suggests that the un-compressed case might not be handled correctly, but I may have just missed it.

I'll try to have a more detailed look soon.

Reply:
Aah yes, you're right. I'll change the condition to something like `HAVE_INDEXED_GZIP and isinstance(file_like, str) and file_like.endswith('.gz')`.